File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    days = [math.ceil((100-p) / s) for p, s in zip(progresses, speeds)]

    idx = 0 
    for i in range(len(days)) :
        if days[idx] < days[i] :      
            answer.append(i - idx)    
            idx = i                   

    answer.append(len(days) - idx)    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    let tempArr = [];

    progresses.forEach((e, i) => {
        let count = 0;

        while(e < 100) {
            e += speeds[i];

            count += 1;
        }

        tempArr.push(count)
    })

    for (let i = 0; i < tempArr.length; i++) {

        let count = 1; 
        for (let j = i + 1; j < tempArr.length; j++) {
            if (tempArr[i] >= tempArr[j]) {
                count += 1;
            } else {
                break;
            }
        }

        answer.push(count);

        i += count - 1;
    }

    return answer;
}